src/server/common/helpers/auth/user-session.js

```diff
@@ -43,44 +40,19 @@ async function createUserSession(request, sessionId) {
 
   const { id, email, displayName, loginHint } = request.auth.credentials.profile
 
-  const { scopes, scopeFlags } = await fetchScopes(
-    request.auth.credentials.token
-  )
-
-  await request.server.session.set(sessionId, {
+  const session = {
     id,
     email,
     displayName,
     loginHint,
     isAuthenticated: request.auth.isAuthenticated,
     token: request.auth.credentials.token,
     refreshToken: request.auth.credentials.refreshToken,
-    isAdmin: scopeFlags.isAdmin,
-    isTenant: scopeFlags.isTenant,
-    scope: scopes,
     expiresIn: expiresInMilliSeconds,
     expiresAt
-  })
-}
-
-/**
- * Update user scope
- * @param {import("@hapi/hapi").Request} request
- * @param {UserSession} userSession
- * @returns {Promise<UserSession> | UserSession}
- */
-async function updateUserScope(request, userSession) {
-  const { scopes, scopeFlags } = await fetchScopes(userSession.token)
-
-  request.logger.debug('User session updated')
-
-  await request.server.session.set(request.state.userSessionCookie.sessionId, {
-    ...userSession,
-    ...(scopeFlags ?? []),
-    scope: scopes ?? []
-  })
+  }
 
-  return await request.getUserSession()
+  await request.server.session.set(sessionId, session)
```

Comment:
IMO - I'd expect a method called `createUserSession` to return the session thats just been created